enumerables.rb

```diff
@@ -0,0 +1,209 @@
+module Enumerable
+  # my_each
+
+  def my_each
+    if block_given?
+      if is_a?(Range)
+        arr = to_a
+        arr.each do |value|
+          yield(value)
+        end
+      else
+        each do |value|
+          yield(value)
+        end
+      end
+      self
+    else
+      to_enum :my_each
+    end
+  end
+
+  # my_each_with_index
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    size.times do |i|
+      yield(to_a[i], i)
+    end
+    self
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    new_a = []
+    my_each do |element|
+      new_a << element if yield(element)
+    end
+    new_a
+  end
+
+  # My_all?
+  def my_all?(arg = nil)
+    arr = to_a if self.class == Range
+    arr = self if self.class != Range
+    if block_given?
+      arr.length.times do |i|
+        return false if yield(arr[i]) == false
+      end
+    else
+      puts 'block was not given'
+      puts "self class is #{arr.class}"
+      puts "argument class is #{arg.class}"
+      if !arg.nil?
+        arr.length.times do |i|
+          if arg.class == Integer
+            unless arr[i].to_s.include?(arg.to_s)
+              puts(" array item #{arr[i]} doesn't equal argument #{arg}, rendering false")
+              return false
+            end
+          elsif arg.class == Class
+            if arg == Numeric || arg == Integer
+              return false unless arr[i].class == Numeric || arr[i].class == Integer
```

Comment:
- [ ] Your my_all is still failing some test cases. For e.g `[1, 2i, 3.14].my_all?(Numeric)` returns `false` while ` [1, 2i, 3.14].my_all?(Numeric)`  returns true. This is failing because by the time your code compares the class of `2i` to `Numeric` it will return false because the class of `2i` will not equal `Numeric` but `Complex`. The reason the original is working and yours does not is because the original is not comparing directly the class of the entities with `Numeric` but is somehow establishing some relationship. If you do this `(2i).class.ancestors` you will get this `[Complex, Numeric, Comparable, Object, Kernel, BasicObject]`. So to get your code to pass, you don't need that extra conditional statement. You should just check if the arg passed is a class and if it is a class just loop over the array and compare the elements of the array with the class. i.e something like this
```
elsif arg.is_a? Class
   return false unless arr[i].class == arg
```